eZ/Bundle/EzPublishCoreBundle/Command/NormalizeImagesPathsCommand.php

```diff
@@ -209,9 +173,117 @@ private function moveFile(
             ]
         );
 
-        $newBinaryFile = $this->ioService->createBinaryFile($binaryCreateStruct);
-        if ($newBinaryFile instanceof BinaryFile) {
-            $this->ioService->deleteBinaryFile($oldBinaryFile);
+        // Before creating a new file validate if the same file doesn't exist already in order to not duplicate files
+        $newBinaryFile = $this->ioService->loadBinaryFileByUri(\DIRECTORY_SEPARATOR . $newPath);
+        if ($newBinaryFile instanceof MissingBinaryFile) {
+            $this->ioService->createBinaryFile($binaryCreateStruct);
+        }
+    }
+
+    protected function updateImagePathsToNormalize(
+        $imageData,
+        array $imagePathsToNormalize
+    ): array {
+        $filePath = $imageData['filepath'];
+        $fieldId = (int)$imageData['contentobject_attribute_id'];
+
+        $finalNormalizedPath = $this->getFinalNormalizedPath(
+            $filePath,
+            $imagePathsToNormalize
+        );
+
+        if ($finalNormalizedPath !== $filePath) {
+            $imagePathsToNormalize[$fieldId] = [
+                'fieldId' => $fieldId,
+                'oldPath' => $filePath,
+                'newPath' => $finalNormalizedPath,
+            ];
         }
+
+        return $imagePathsToNormalize;
+    }
+
+    protected function getFinalNormalizedPath(
```

Comment:
This class is final, so `protected` scope does not make sense.

```suggestion
    private function getFinalNormalizedPath(
```